[TABLE] Some UI adjusments

KevinFabre-ods opened the pull request:
## Summary

The goal for this PR is to fix some visual bugs for the table component

(Internal for Opendatasoft only) Associated Shortcut tickets: 
- [sc-50076](https://app.shortcut.com/opendatasoft/story/50076)
- [sc-50178](https://app.shortcut.com/opendatasoft/story/50178)
- [sc-50164](https://app.shortcut.com/opendatasoft/story/50164)

### Changes
- Display left ellipsis on the pagination
- Table columns take all available space in the card
- Add a label object in the table options API. To label pagination buttons and how records are named
- Numbering label no longer breaks on a new line
- Wrap the pagination elements when width is small


## Review checklist

- [x] Description is complete
- [x] Commits respect the [Conventional Commits Specification](https://github.com/opendatasoft/ods-dataviz-sdk/blob/main/CONTRIBUTING.md#commit-messages)
- [x] 2 reviewers (1 if trivial)
- [ ] Tests coverage has improved
- [ ] Code is ready for a release on NPM

KevinFabre-ods commented on packages/visualizations-react/stories/Table/Pagination.stories.tsx:
```diff
@@ -104,5 +103,6 @@ CustomStyle.args = {
     options: {
         ...paginatedOptions,
         unstyled: true,
+        labels: undefined,
```
If `labels` not defined, `Numbering` component only shows `1-5/xxx`

etienneburdet commented on packages/visualizations/src/components/Table/Pagination/Pagination.svelte:
```diff
@@ -40,7 +40,9 @@
     .pagination {
         display: flex;
-        justify-content: space-between;
+        flex-wrap: wrap;
+        justify-content: space-evenly;
+        gap: var(--spacing-100);
```
Better for vertical display indeed, looks a bit strange with the spacer but it will look good with the page size indeed 👌

KevinFabre-ods commented:
> All good to me. I just left a comment to add a comment in the code (the one you left as a github comment), because it took me a minute to get. Else it all works good.

@etienneburdet  Since your review, I merged main into this branch to resolve conflicts with the latest branch we merged (https://github.com/opendatasoft/ods-dataviz-sdk/pull/232)

Nothing changed much, I moved the `labels` property directly in the pagination types. Also I updated a bit the pagination layout with the use of grid and `container` query to handle small size of the table.

KevinFabre-ods commented:
@etienneburdet Chromatic changes must be accepted though

KevinFabre-ods commented:
With https://github.com/opendatasoft/ods-dataviz-sdk/pull/238/commits/b09a20afaba988ce444ea6552203b2f731f5e2e5 pagination elements can be targeted via global classes allowing simple CSS customization. 

Some stories might be impacted

cybervinvin commented on package-lock.json:
This file has no link with the PR, right ?

KevinFabre-ods replied:
yes indeed not link

cybervinvin reviewed (approved):
Bravo for the use of css-grid in the pagination 👏🏼
I have few small non-blocking comments to correct (or not 😉 )
LGTM 🚀

KevinFabre-ods commented on packages/visualizations-react/stories/Table/Table.stories.tsx:
```diff
@@ -36,7 +36,22 @@ const CustomStyleTemplate: ComponentStory<typeof Table> = args => (
 CustomStyle.args = {
     data,
-    options,
+    options: {
+        ...options,
+        pagination: {
+            current: 1,
+            totalRecords: value.length,
+            recordsPerPage: 5,
+            onPageChange: () => {},
+            pageSizeSelect: {
+                options: [
+                    { label: '5 per page', value: 5 },
+                    { label: '10 per page', value: 10 },
+                ],
+                onChange: () => {},
+            },
+        },
+    },
```
This story is to show that every part of the table can be override with CSS.
Before this PR we didn't show the pagination for this story. I think we should

We will clean up the table stories when we have less work to do. I agreed we can do better

etienneburdet commented on packages/visualizations/src/components/Pagination/Button.svelte:
```diff
@@ -35,11 +36,11 @@
     :global(.ods-dataviz--default) {
         button {
-            color: '#142E7B';
+            color: '#000';
         }
 
         button:disabled {
-            color: '#C6CBDE';
+            color: '#DEE5EF';
```
Historically, the SDK is supposed to be "neutral" and we apply color in the platform. They could be "default" through the `Card` component thought.
